tools/ossf_best_practices/passing_criteria.md

```diff
@@ -0,0 +1,397 @@
+
+**N/A**
+
+> The security mechanisms within the software produced by the project MUST generate all cryptographic keys and nonces using a cryptographically secure random number generator, and MUST NOT do so using generators that are cryptographically insecure.
+
+**Met**
+
+## Secured delivery against man-in-the-middle (MITM) attacks
+
+> The project MUST use a delivery mechanism that counters MITM attacks. Using https or ssh+scp is acceptable.
+
+**Met**
+
+> A cryptographic hash (e.g., a sha1sum) MUST NOT be retrieved over http and used without checking for a cryptographic signature.
+
+**Met**
+https://github.com/nodejs/node#verifying-binaries
+
+
+## Publicly known vulnerabilities fixed
+
+> There MUST be no unpatched vulnerabilities of medium or higher severity that have been publicly known for more than 60 days.
+
+**Met**
+
+> Projects SHOULD fix all critical vulnerabilities rapidly after they are reported.
+
+**Met**
+
+
+## Other security issues
+
+> The public repositories MUST NOT leak a valid private credential (e.g., a working password or private key) that is intended to limit public access
+
+**Met**
+
+# Analysis
+
+## Static code analysis
+
+> At least one static code analysis tool (beyond compiler warnings and "safe" language modes) MUST be applied to any proposed major production release of the software before its release, if there is at least one FLOSS tool that implements this criterion in the selected language.
+
+**Met**
+https://scan.coverity.com/projects/node-js
+
+> It is SUGGESTED that at least one of the static analysis tools used for the static_analysis criterion include rules or approaches to look for common vulnerabilities in the analyzed language or environment.
+
+**Met**
+
+> All medium and higher severity exploitable vulnerabilities discovered with static code analysis MUST be fixed in a timely way after they are confirmed.
+
+**Met**
+
+> It is SUGGESTED that static source code analysis occur on every commit or at least daily.
+
+**Unmet**
+
+## Dynamic code analysis
+
+> It is SUGGESTED that at least one dynamic analysis tool be applied to any proposed major production release of the software before its release.
+
+**Unmet**
+Infrastructure for running several different dynamic analysis tools is provided by the project. See: https://github.com/nodejs/node/tree/master/tools
+
+
+> It is SUGGESTED that if the software produced by the project includes software written using a memory-unsafe language (e.g., C or C++), then at least one dynamic tool (e.g., a fuzzer or web application scanner) be routinely used in combination with a mechanism to detect memory safety problems such as buffer overwrites. If the project does not produce software written in a memory-unsafe language, choose "not applicable" (N/A).
+
+**Met**
+valgrind for c++
```

Comment:
I'm curious if this counts given that it says "one dynamic tool (...) **in combination with a mechanism to detect memory safety problems**". Valgrind does qualify as the latter, but does it also count as the dynamic tool? It does not replace mechanisms such as fuzzing.

Reply:
Good point @tniessen! I think that qualifies based on this definition:

> Valgrind is an instrumentation framework for building dynamic analysis tools. There are Valgrind tools that can automatically detect many memory management and threading bugs, and profile your programs in detail. You can also use Valgrind to build new tools.
[Source](https://valgrind.org/)